source/crud/read-operations/cursor.txt

```diff
@@ -0,0 +1,174 @@
+=========================
+Access Data From a Cursor
+=========================
+
+.. default-domain:: mongodb
+
+Read operations that return multiple documents do not immediately return
+all values matching the query. Because a query can potentially match
+very large sets of documents, these operations rely upon an
+object called a :node-docs:`cursor <api/Cursor.html>`
+that fetches documents in batches to reduce both memory consumption and
+network bandwidth usage. Cursors are highly configurable and offer
+multiple interaction paradigms for different use cases.
+
+The following functions directly return cursors:
+
+- ``Collection.find()``
+
+- ``Collection.aggregate()``
+
+- ``Collection.listIndexes()``
+
+- ``Db.aggregate()``
+
+- ``Db.listCollections()``
+
+Other methods, like ``Collection.findOne()`` and ``Collection.watch()``,
+use cursors to generate results but do not directly expose those
+cursors to the method caller.
+
+Cursor Paradigms
+----------------
+
+You can work with cursors using a number of **cursor paradigms**.
+Many use cases only require one document at a time, so many of these
+cursor paradigms take advantage of this by abstracting away network
+accesses and caching behind individual document accesses. Some use cases
+require all documents at once or only a count of documents, so paradigms
+exist for those needs as well.
+
+.. warning::
+
+   Do not combine different cursor paradigms on a single cursor.
+   Operations such as ``hasNext()``, ``forEach()``, and ``toArray()``
+   each predictably modify the original cursor. If you mix these calls
+   on a single cursor, you may receive unexpected results.
+
+.. warning::
+
+   Because asynchronous calls directly modify the cursor, executing
+   asynchronous calls on a single cursor simultaneously can also cause
+   undefined behaviour. As a result, always wait for the previous
+   asynchronous operation to complete before running another
+   asynchronous operation.
+
+For Each Functional Iteration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can pass a function to the :node-docs:`forEach()
+<api/Cursor.html#forEach>` method of any cursor to iterate through
+cursor results in a functional style:
+
+.. literalinclude:: /code-snippets/crud/cursor.js
+   :language: javascript
+   :start-after: start foreach cursor example
+   :end-before: end foreach cursor example
+   :dedent: 4
+
+Asynchronous Iteration
+~~~~~~~~~~~~~~~~~~~~~~
+
+Cursors implement the AsyncIterator interface, which allows you to
+use Cursors in more traditional ``for``...``await`` loops:
```

Comment:
Nitpick 1: We should stay consistent with capitalization of "cursor".
Nitpick 2: We can probably leave out adjectives like "traditional" for brevity since it does not add much value for readers that either know or do not know what they are.

Reply:
Both good points.
- standardized lowercase cursor
- removed "traditional" (and on the look out for more cases of extraneous adjectives that don't make docs any better)